.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -0,0 +1,1622 @@
+                yearStr = HandleWithHalfYear(match, yearStr, out bool hasHalf, out bool isFirstHalf);
+
+                if (this.config.IsYearOnly(yearStr))
+                {
+                    yearStr = yearStr.Substring(0, yearStr.Length - 1);
+                }
+
+                if (yearStr.Length == 1)
+                {
+                    return ret;
+                }
+
+                var year = ConvertChineseToInteger(yearStr);
+
+                return HandleYearResult(ret, hasHalf, isFirstHalf, year);
+            }
+
+            return ret;
+        }
+
+        private string HandleWithHalfYear(ConditionalMatch match, string text, out bool hasHalf, out bool isFirstHalf)
+        {
+            var firstHalf = match.Groups["firstHalf"].Value;
+            var secondHalf = match.Groups["secondHalf"].Value;
+
+            hasHalf = false;
+            isFirstHalf = !string.IsNullOrEmpty(firstHalf);
+
+            if (isFirstHalf || !string.IsNullOrEmpty(secondHalf))
+            {
+                var halfText = isFirstHalf ? firstHalf : secondHalf;
+                text = text.Substring(0, text.Length - halfText.Length);
+                hasHalf = true;
+            }
+
+            return text;
+        }
+
+        private DateTimeResolutionResult HandleYearResult(DateTimeResolutionResult ret, bool hasHalf, bool isFirstHalf, int year)
+        {
+            if (year < 100 && year >= this.config.TwoNumYear)
+            {
+                year += 1900;
+            }
+            else if (year < 100 && year < this.config.TwoNumYear)
+            {
+                year += 2000;
+            }
+
+            var beginDay = DateObject.MinValue.SafeCreateFromValue(year, 1, 1);
+            var endDay = DateObject.MinValue.SafeCreateFromValue(year + 1, 1, 1);
+
+            ret.Timex = DateTimeFormatUtil.LuisDate(year);
+
+            if (hasHalf)
+            {
+                if (isFirstHalf)
+                {
+                    endDay = DateObject.MinValue.SafeCreateFromValue(year, 7, 1);
+                }
+                else
+                {
+                    beginDay = DateObject.MinValue.SafeCreateFromValue(year, 7, 1);
+                }
+
+                ret.Timex = TimexUtility.GenerateDatePeriodTimex(beginDay, endDay, DatePeriodTimexType.ByMonth);
+            }
+
+            ret.FutureValue = ret.PastValue = new Tuple<DateObject, DateObject>(beginDay, endDay);
+            ret.Success = true;
+
+            return ret;
+        }
+
+        // parse entities that made up by two time points
+        private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject referenceDate)
+        {
+            var ret = new DateTimeResolutionResult();
+            var er = this.config.DateExtractor.Extract(text, referenceDate);
+            if (er.Count < 2)
+            {
+                er = this.config.DateExtractor.Extract("on " + text, referenceDate);
```

Comment:
"on "? Was this already in the Chinese code before?

Reply:
yes, replaced with TokenBeforeDate as in standard implementation